frontend/components/Badge/index.jsx

```diff
@@ -1,25 +1,242 @@
+    ...badgeContainerStylingSliders,
+  }).toString(),
+  containerSlidersTopRight: css({
+    display: 'flex',
+    position: 'absolute',
+    alignItems: 'end',
+    flexDirection: 'column-reverse',
+    top: 5,
+    right: 5,
+    zIndex: 1000,
+    ...badgeContainerStylingSliders,
+  }).toString(),
+  containerSlidersBottomLeft: css({
+    display: 'flex',
+    flexDirection: 'column-reverse',
+    alignItems: 'start',
+    position: 'absolute',
+    bottom: productListType === 'productSlider' && (productListSubType === 'widgets' || productListSubType === 'upselling') ? 100 : 120,
+    left: 5,
+    ...badgeContainerStylingSliders,
+  }).toString(),
+  containerSlidersBottomRight: css({
+    display: 'flex',
+    flexDirection: 'column-reverse',
+    position: 'absolute',
+    alignItems: 'end',
+    bottom: productListType === 'productSlider' && (productListSubType === 'widgets' || productListSubType === 'upselling') ? 100 : 120,
+    right: 5,
+    ...badgeContainerStylingSliders,
+  }).toString(),
+  containerLists: css({
+    display: 'flex',
+    position: 'absolute',
+    flexDirection: 'column-reverse',
+    alignItems: 'start',
+    top: 5,
+    left: 5,
+    zIndex: 1000,
+    ...badgeContainerStylingLists,
+  }).toString(),
+  containerListsTopRight: css({
+    display: 'flex',
+    position: 'absolute',
+    alignItems: 'end',
+    flexDirection: 'column-reverse',
+    top: 5,
+    right: 5,
+    zIndex: 1000,
+    ...badgeContainerStylingLists,
+  }).toString(),
+  containerListsBottomLeft: css({
+    display: 'flex',
+    flexDirection: 'column-reverse',
+    alignItems: 'start',
+    position: 'absolute',
+    bottom: 120,
+    left: 5,
+    ...badgeContainerStylingLists,
+  }).toString(),
+  containerListsBottomRight: css({
+    display: 'flex',
+    flexDirection: 'column-reverse',
+    position: 'absolute',
+    alignItems: 'end',
+    bottom: 120,
+    right: 5,
+    ...badgeContainerStylingLists,
+  }).toString(),
+  badge: css({
+    ...badgeStylingPdp,
+  }).toString(),
+  badgeSliders: css({
+    ...badgeStylingSliders,
+  }).toString(),
+  badgeLists: css({
+    ...badgeStylingLists,
+  }).toString(),
+});
 
 /**
  * @param {Array} badgeInfo Array of badge image urls
  * @returns {JSX}
  */
-const CardBadge = ({ badgeInfo }) => {
-  const images = badgeInfo.map((image, index) => (
-    <img className={styles.badge} src={image} alt="" key={index.toString()} />
-  ));
-
-  return (
-    <div className={styles.container}>
-      {images}
-    </div>
-  );
+const CardBadge = ({ badgeInfo, badgePosition }) => {
+  const { productListType, productListSubType } = useProductListEntry();
+
+  console.log('---productListType----', productListType);
+  console.log('---productListSubType----', productListSubType);
+
+  if (!badgeInfo || badgeInfo.length === 0) {
+    return null;
+  }
+
+  const styles = getStyles(productListType, productListSubType);
+
+  let images;
+
+  if (productListType === 'productSlider') {
```

Comment:
to avoid re-computation at every render, please move this inside `useMemo`

```js
  const images = useMemo(() => {
    if (!Array.isArray(badgeInfo) || badgeInfo.length === 0) {
      return [];
    }

    if (productListType === 'productSlider') {
      return badgeInfo.map((image, index) => (
        <img className={stylesNew.badgeSliders} src={image} alt="" key={index.toString()} />
      ));
    } 
    
    if (productListType === 'productGrid') {
      return badgeInfo.map((image, index) => (
        <img className={stylesNew.badgeLists} src={image} alt="" key={index.toString()} />
      ));
    }
    return badgeInfo.map((image, index) => (
      <img className={stylesNew.badge} src={image} alt="" key={index.toString()} />
    ));
  }, [badgeInfo, productListType]);
```